spec/validation/configure_spec.rb

```diff
@@ -34,244 +41,264 @@
           type: 'integer',
           default: 10,
         },
-        boolean_true: {
+        {
+          identifier: 'boolean_true',
           question: 'Can I have a boolean true (/yes) default?',
           type: 'boolean',
-          default: 'yes',
+          default: true,
         },
-      },
+      ],
 
-      group: {
-        string_question: {
+      group: [
+        {
+          identifier: 'string_question',
           question: 'Am I a string question without a type but with a default?',
           default: 'yes I am a string',
         },
-        integer_question: {
+        {
+          identifier: 'integer_question',
           question: 'Am I a integer question without a default?',
           type: 'integer',
         },
-        boolean_false: {
+        {
+          identifier: 'boolean_false',
           question: 'Can I have a boolean false (/no) default?',
           type: 'boolean',
-          default: 'no',
+          default: false,
         },
-      },
+      ],
 
-      node: {
-        string_question: {
+      node: [
+        {
+          identifier: 'string_question',
           question: 'Am I a string question with a type and default?',
           type: 'string',
           default: 'yes I am a string',
         },
-        string_empty_default: {
+        {
+          identifier: 'string_empty_default',
           question: 'My default is a empty string?',
           default: '',
         },
-      },
+      ],
 
-      self: {},
+      self: [
+        {
+          identifier: 'choice question',
+          question: 'Are all my choices strings?',
+          choice: [
+            'choice1',
+            'choice2',
+            'choice3',
+          ],
+          default: 'choice1',
+        },
+        {
+          identifier: 'choice_question_no_default',
+          question: 'Are all my choices strings without a default?',
+          choice: [
+            'choice1',
+            'choice2',
+            'choice3',
+          ],
+        },
+      ],
     }
   end
 
-  def build_validator(my_hash = {})
-    allow(Metalware::Data).to receive(:load).and_return(my_hash)
-    Metalware::Validation::Configure.new('path/has/been/mocked')
+  context 'without a hash input' do
+    it 'loads configure file from the repo' do
+      data = { data: 'I am the configure data' }
+      Metalware::Data.dump(file_path.configure_file, data)
+      v = Metalware::Validation::Configure.new(config)
+      expect(v.send(:raw_data)).to eq(data)
+    end
+  end
+
+  context 'with a hash input' do
+    it 'uses the has as the data input' do
+      v = Metalware::Validation::Configure.new(config, correct_hash)
+      expect(v.send(:raw_data)).to eq(correct_hash)
+    end
   end
 
   def run_configure_validation(my_hash = {})
-    validator = build_validator(my_hash)
-    validator.validate.messages
+    Metalware::Validation::Configure.new(config, my_hash).data
+  end
+
+  def expect_configure_error(my_hash, msg_regex)
```

Comment:
A more descriptive name for this might be `expect_validation_failure` - what do you think?

Reply:
fixed: https://github.com/alces-software/metalware/pull/185/commits/bb4f11c336bedf9e1138f9320e151fcca5e974c4